core/src/main/java/io/th0rgal/oraxen/utils/inventories/ItemsView.java

```diff
@@ -129,30 +129,41 @@ private record GuiItemSlot(ItemStack itemStack, Integer slot) {
+        //Material of category itemstack. if no material is set, set it to the first item of the category
+        Optional<String> icon = Optional.ofNullable(settings.getString(String.format("oraxen_inventory.menu_layout.%s.icon", fileName)));
         String displayName = ItemParser.parseComponentItemName(settings.getString(String.format("oraxen_inventory.menu_layout.%s.displayname", fileName), "<green>" + file.getName()));
-        try {
-            itemStack = new ItemBuilder(OraxenItems.getItemById(icon).getReferenceClone())
-                    .addItemFlags(ItemFlag.HIDE_ATTRIBUTES)
-                    .setDisplayName(displayName)
-                    .setLore(new ArrayList<>())
-                    .build();
-        } catch (final Exception e) {
-            try {
-                itemStack = new ItemBuilder(Material.getMaterial(icon.toUpperCase()))
+
+        itemStack = icon
+                .map(OraxenItems::getItemById)
+                .map(ib -> new ItemBuilder(ib.getReferenceClone())
                         .addItemFlags(ItemFlag.HIDE_ATTRIBUTES)
                         .setDisplayName(displayName)
-                        .build();
-            } catch (final Exception ignored) {
-                itemStack = new ItemBuilder(Material.PAPER)
-                        .setDisplayName(displayName)
-                        .build();
-            }
-        }
+                        .setLore(new ArrayList<>())
+                        .build())
+                .orElseGet(() -> icon
+                        .map(String::toUpperCase)
+                        .map(Material::getMaterial)
+                        .map(material -> new ItemBuilder(material)
+                                .addItemFlags(ItemFlag.HIDE_ATTRIBUTES)
+                                .setDisplayName(displayName)
+                                .build())
+                        .orElseGet(() -> {
+                            try {
+                                return new ItemBuilder(OraxenItems.getMap().get(file).entrySet().iterator().next().getValue().getReferenceClone())
+                                        .setDisplayName(displayName)
+                                        .build();
+                            } catch (Exception ignored) {
+                                return new ItemBuilder(Material.PAPER)
+                                        .setDisplayName(displayName)
+                                        .build();
+                            }
+                        })
+                );
 
         // avoid possible bug if isOraxenItems is available but can't be an itemstack
         if (itemStack == null) itemStack = new ItemBuilder(Material.PAPER).setDisplayName(displayName).build();
         int slot = settings.getInt(String.format("oraxen_inventory.menu_layout.%s.slot", Utils.removeExtension(file.getName())), 0) - 1;
+        System.out.println("Slot: " + slot + " for item: " + itemStack.displayName());
```

Comment:
Remove this, if debugging use Logs.debug, never System.out.println
Debug is also kept to a minimum in releases

Reply:
Ups